src/client.js

```diff
@@ -166,9 +166,18 @@ export function getImages(system, id) {
                                     images[info.Id] = Object.assign(images[info.Id], parseImageInfo(info));
                                     images[info.Id].isSystem = system;
                                 }
-                                resolve(images);
                             })
                             .catch(reject);
+
+                    for (const image of immages || []) {
```

Comment:
This cannot be like this here as it may execute and `resolve()` before previous promises are done, thus we might miss some information. You need to do this in sync with previous promises.

Reply:
This also seems to be a performance issue when loading images, so we should either use:

https://docs.podman.io/en/latest/_static/api.html#tag/images/operation/ImageListLibpod which already has History but lacks the image size. Or we load the History on demand (which makes a lot more sense) when opening the tab. We do the same thing with the Container integration tab.

Reply:
Yes, I think on demand would be the best option. Storing history for each image in memory is rather useless as I don't expect a load of people looking into it. And waiting for second when toggling the tab seems like ideal solution.